github_visualization/github_stats.py

```diff
@@ -0,0 +1,289 @@
+            r_json = response.json()
+            basic_stats["repo_name"] = r_json["name"]
+            basic_stats["repo_description"] = r_json["description"]
+            basic_stats["is_private"] = r_json["private"]
+            basic_stats["html_url"] = r_json["html_url"]
+            basic_stats["stars"] = r_json["stargazers_count"]
+            basic_stats["stars_url"] = r_json["html_url"] + "/stargazers"
+            basic_stats["watchers"] = r_json["watchers_count"]
+            basic_stats["forks"] = r_json["forks_count"]
+            basic_stats["forks_url"] = r_json["html_url"] + "/network"
+            basic_stats["open_issues"] = r_json["open_issues_count"]
+            basic_stats["subscribers"] = r_json["subscribers_count"]
+
+            return basic_stats
+        except:
+            raise
+
+    def fetch_contributor_stats(self):
+        ''' Fetch stats of contributors
+
+        Returns
+        -------
+        contributor_stats : dict
+            A dictionary containting contributor statistics. For Example:
+            { 'response_code': 200,
+              'total_contributors': 50,
+              'total_commits': 6031,
+              'contributors': [ {
+                                "user_name": "Garyfallidis"
+                                "avatar_url": "https://avatars.githubusercontent.com/u/134276?v=3",
+                                "html_url": "https://github.com/Garyfallidis",
+                                "total_commits": 1389,
+                                "total_additions": 116712,
+                                "total_deletions": 70340,
+                                "weekly_commits": [
+                                          {
+                                            "w": "1367712000",
+                                            "a": 6898,
+                                            "d": 77,
+                                            "c": 10
+                                          },
+                                        ]
+                                },
+                            ]
+              'cumulative_contributors': [
+                                (
+                                    1367712000, // timestamp
+                                    20 // total number of contributions
+                                       // upto this time
+                                ),
+                            ]
+            }
+
+        '''
+        contributor_stats = {}
+        contributor_stats["contributors"] = []
+
+        url = self.baseurl + "/stats/contributors"
+        try:
+            response = requests.get(url)
+            contributor_stats["response_code"] = response.status_code
+
+            # check response code
+            if response.status_code != 200:
+                return contributor_stats
+
+            # parse response
+            r_json = response.json()
+            contributor_stats["total_contributors"] = len(r_json)
```

Comment:
Well, I'm not really sure. It's just that, in the documentation of the current function, there is the 'cumulative_contributors' field in the JSON. Wouldn't this give you exactly this information, instead of using the len(...)? Just wondering.